appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/EJBContainerTransactionManager.java

```diff
@@ -388,37 +380,36 @@ protected void useClientTx(Transaction prevTx, EjbInvocation inv) {
                     container.afterBegin(context);
                 }
             } catch (Exception ex) {
+                final TransactionRolledbackLocalException toThrow = new TransactionRolledbackLocalException("", ex);
                 try {
                     transactionManager.setRollbackOnly();
-                } catch ( Exception e ) {
-					//FIXME: Use LogStrings.properties
-					_logger.log(Level.FINEST, "", e);
+                } catch (Exception e) {
+                    toThrow.addSuppressed(e);
 				}
-                throw new TransactionRolledbackLocalException("", ex);
+                throw toThrow;
```

Comment:
It is tiny change - and it passed TCK at night. Except one test broken also in master. That's my next issue.

Reply:
These changes are because EJB container sometimes drops exceptions. As you have seen, it will need much more work, but I'm pretty sure this will not break anything.

Reply:
I don't say don't fix what is broken, just split a systematic cleanup from a goal oriented bugfix. I know that cleaning when looking for a fix is also needed and a good way to find and fix problems but usually such changes are somewhat limited in scope and do not change 20 classes each having hundreds of changing lines.